### ISSUE TITLE
Injecting Services?

### ISSUE THREAD

**Issue by diit** [AngularClass/NG6-starter#85]
How to go about injecting a service into a controller, I have tried ```.service('tickerTapeService', tickerTapeService)``` as well as importing it in the controller itself. Can't seem to find solution or documentation.

**Comment by arturoromerotcs** [comment]
Here is an example of a service 


```javascript 
import {tickerTapeService} from './tickerTapeService';
import angular from 'angular';

//I like to register service in module to keep Angular module system, which will be helpful when unit testing and mocking data.
//Service Shared Module
export const shared = angular.module('shared', [])
.service('tickerTapeService', tickerTapeService);

//Service 
const tickerTapeService = ($http) => {
  let data = [];

  const get = () => {
    return $http.get(`${API.url}/restful/path`)
      .then(resp => data = resp.data);
  };

  const getState = () => {
    return data;
  };

  return {get, getState};
};

//Inject $http provider for minification.
tickerTapeService.$inject = ['$http'];

//export service module after injection.
export {tickerTapeService};


//Controller
class someController {
//Inject the tickerTapeService service into the Controller's constructor function.   
constructor(tickerTapeService) {
//someController now has tickerTapeService as a DI. 
    this.tickerTapeService = tickerTapeService;
    this.getData();
  }

  getData() {
    this.tickerTapeService.get()
      .then(() => {
        this.data = this.tickerTapeService.getState();
      });
  }
}

//Inject service for minification.
someController.$inject = ['tickerTapeService'];

export {someController};

//App Module
angular.module('app', [
  uiRouter,
  ngAnimate,
  // all angular modules have a name property which is a string representation of the name you set the on the module
  shared.name
])

```

**Comment by fesor** [comment]
@diit I prepared a small example of application using services: [ng6-todomvc](https://github.com/AngularClass/NG6-todomvc-starter).

It also contains an example of how to test them.

**Comment by fesor** [comment]
> How to go about injecting a service into a controller

Please read this section for more details: http://blog.thoughtram.io/angularjs/es6/2015/01/23/exploring-angular-1.3-using-es6.html#angular-and-es-modules

It describes how to use ES6 modules with angular modules.

I'm closing this, if you have any other question, please continue in #60

**Comment by arturoromerotcs** [comment]
I found a bug and I have updated my code.  My service is a factory as I was returning an object from the tickerTapeService.

@fesor Because I'm using a .factory() and not a  .serivce(). A service is a "newable" function and uses this. if I was actually using a service I could use a class.  Since I'm using a factory() return an object so a function is enough no need for a class. 

After dabbing with React and Unidirectional data I found client side state to be useful, just a preference. :+1:
